File: framework/src/play-test/src/main/java/play/test/FakeApplication.java
package play.test;

import java.io.*;
import java.util.*;

import play.libs.*;

/**
 * A Fake application.
 */
public class FakeApplication {

    final play.api.test.FakeApplication wrappedApplication;

    /**
     * A Fake application.
     *
     * @param path The application path
     * @param classloader The application classloader
     * @param additionalConfiguration Additional configuration
     * @param additionalPlugins Additional plugins
     */
    public FakeApplication(File path, ClassLoader classloader, Map<String, ? extends Object> additionalConfiguration, List<String> additionalPlugins) {
        wrappedApplication = new play.api.test.FakeApplication(
                path,
                classloader,
                Scala.toSeq(additionalPlugins),
                Scala.<String>emptySeq(),
                Scala.asScala((Map<String, Object>)additionalConfiguration)
                );
    }

    public play.api.test.FakeApplication getWrappedApplication() {
        return wrappedApplication;
    }

}

File: framework/src/play/src/test/scala/play/api/FakeApplication.scala
/*
 * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
 */
package play.api

import akka.stream.ActorMaterializer
import play.api.http.{ NotImplementedHttpRequestHandler, DefaultHttpErrorHandler }
import play.api.libs.concurrent.ActorSystemProvider
import java.io.File

/**
 * Fake application as used by Play core tests.  This is needed since Play core can't depend on the Play test API.
 * It's also a lot simpler, doesn't load default config files etc.
 */
case class FakeApplication(config: Map[String, Any] = Map(),
    path: File = new File("."),
    mode: Mode.Mode = Mode.Test,
    override val global: GlobalSettings = DefaultGlobal,
    plugins: Seq[Plugin.Deprecated] = Nil) extends Application {
  val classloader = Thread.currentThread.getContextClassLoader
  lazy val configuration = Configuration.from(config)
  private val lazyActorSystem = ActorSystemProvider.lazyStart(classloader, configuration)
  def actorSystem = lazyActorSystem.get()
  lazy val materializer = ActorMaterializer()(actorSystem)
  def stop() = lazyActorSystem.close()
  val errorHandler = DefaultHttpErrorHandler
  val requestHandler = NotImplementedHttpRequestHandler
}
